[Feat] 기업·증권사 가입 신청 승인/거절 흐름 구현(#13)

ggj0228 opened the pull request:
## 📋 작업 개요
  기업·증권사 가입 신청을 파일 업로드 기반으로 받아 승인/거절 흐름을 구현했습니다.
  <br/>

  ## 🔧 작업 상세

  - 기업/증권사 가입 DTO를 multipart/form-data로 전환하고, 필수 서류 파일을 S3에 업로드한 뒤 URL을 저장하도록 서비스 로직 수정
  - Admin 엔티티에 기업·증권사 연관관계 및 상태 전환 메서드(changeStatus) 추가, 공통 가입 서비스로 중복 검증·비밀번호 암호화 통합
  - 기업/증권사 엔티티에 rejectReason·approve()/reject() 추가하여 승인/거절 상태 추적
  - EXCHANGE 전용 승인 컨트롤러·서비스 구성(목록/상세/승인/거절)과 요약/상세 DTO 작성, 대표 관리자(/admin/approvals/me) 전용 조회 API 구현
  - X-User-Role, X-User-Id 헤더 검증으로 역할별 접근을 제한하고, 예외 시 공통 예외 포맷으로 응답
  <br/>

  ## 📌 이슈 번호
close #13 
  <br/>

  ## 🧪 테스트 결과
기업 목록 조회화면입니다
<img width="825" height="571" alt="image" src="https://github.com/user-attachments/assets/f642a4de-ad6c-4c74-a949-bab26efe95c0" />

증권사 목록조회입니다.
<img width="869" height="557" alt="image" src="https://github.com/user-attachments/assets/547266de-03ce-4c63-8dc3-a0db2dfd07a4" />

기업 및 증권사 가입 요청 허가 후 , PENDING  -> ACTIVE
기업 및 증권사 가입 요청 거절 후 , PENDING  -> REJECTED
<img width="885" height="483" alt="image" src="https://github.com/user-attachments/assets/2656f4ac-a380-42a1-a485-5c6ff6ef9918" />

기업 및 관리자는 가입 신청 상태 조회 가능 
<img width="853" height="629" alt="image" src="https://github.com/user-attachments/assets/b4f3f446-c196-4626-91fb-163edfbefeec" />


  <br/>

  ## ⚠️ 참고사항

  - reject_reason 컬럼 및 REJECTED enum 값이 DB에 반영되어야 합니다.
  - Postman 등 클라이언트에서 CSRF 토큰 및 X-User-Role 헤더를 반드시 포함해야 승인 API 호출이 성공합니다.
  <br/>

  ## 👥 리뷰어 지정
  @solidify-d  @jinnn12  
  <br/>

  ## ✅ PR Checklist

  - [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
  - [x] PR 전 develop 브랜치로부터 Pull 후 충돌 여부를 확인/처리했습니다.
  - [x] 코드에 대한 테스트를 진행 하였으며, 결과에 대한 자료를 첨부하였습니다.
  - [x] 최소 2명의 리뷰어를 지정했습니다.

ggj0228 committed:
Feat: 기업·증권사 가입 검증 및 엔드포인트 정리

- 기업·증권사 가입 DTO·서비스·저장소 구성하고 공통 관리자 생성 로직 추가
- 가입 경로 /auth/signup/**를 신설하고 CSRF 필터 예외 목록 정리
- Admin 엔티티에 소속 정보와 PENDING 기본값을 반영하고 중복 검증 주석 보강

Refs: #13

ggj0228 committed:
Feat: 기업·증권사 가입 승인 흐름 구현

- 기업·증권사 신청 DTO를 multipart로 바꾸고 S3 업로드 후 URL 저장
- Admin 엔티티 및 서비스에 신청 연관·상태 전환 로직 추가
- 승인/거절용 컨트롤러·서비스·요약/상세 DTO 구성, 역할별 접근 제어

Refs: #13

solidify-d commented on mkx-platform/src/main/java/com/beyond/MKX/domain/admin/controller/AdminApprovalController.java:
```diff
@@ -0,0 +1,129 @@
+
+    @GetMapping("/corporations")
+    public ResponseEntity<?> listPendingCorporations(@RequestHeader("X-User-Role") String roleHeader) {
+        requireRole(roleHeader, Role.EXCHANGE);
```
엔드포인트마다 requireRole 함수를 호출하게 되는데 조금 더 효율적인 방법이 있지 않을까, 하는 생각이 듭니다.

다만 필수적인 사항은 아니니 지금은 넘어가되 추후 시간이 허락한다면 디벨롭 해도 좋을것 같아요.

solidify-d commented on mkx-platform/src/main/java/com/beyond/MKX/domain/admin/controller/AdminApprovalController.java:
```diff
@@ -0,0 +1,129 @@
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/admin/approvals")
```
approval-requests엔드포인트가 조금 더 적절해 보이나, 우영님의 판단에 맡기겠습니다.

solidify-d commented on mkx-platform/src/main/java/com/beyond/MKX/domain/admin/dto/CorporationSignUpApprovalDto.java:
```diff
@@ -0,0 +1,53 @@
+
+@Getter
+@Builder
+public class CorporationSignUpApprovalDto {
```
CorporationSignUpApprovalDetailDto로 Dto명 수정 부탁드릴게요~!
아래에 Securityfirm도 동일한 수정 부탁드립니다.

solidify-d commented on mkx-platform/src/main/java/com/beyond/MKX/domain/admin/dto/MySignUpStatusDto.java:
```diff
@@ -0,0 +1,26 @@
+@Getter
+@Builder
+public class MySignUpStatusDto {
+    private String type; // CORPORATION or SECURITIES_FIRM
```
ENUM 사용 부탁드리겠습니다.

ggj0228 committed:
Refactor: 관리자 가입신청 DTO 구조 정비

- MySignUpStatusDto에 SignUpType enum 도입
- 기업/증권사 가입신청 상세 DTO 명칭을 DetailDto로 변경
- 승인 조회 서비스와 컨트롤러를 새 DTO와 URL(/admin/approval-requests)로 정리

Refs #13
